modules/apps/object/object-service/src/main/java/com/liferay/object/internal/search/spi/model/query/contributor/ObjectEntryKeywordQueryContributor.java

```diff
@@ -253,15 +254,51 @@ private void _contribute(
 
 			queryConfig.addHighlightFieldNames(fieldName);
 		}
-		else if (Objects.equals(objectField.getDBType(), "BigDecimal")) {
+		else if (Objects.equals(
+					objectField.getBusinessType(),
+					ObjectFieldConstants.BUSINESS_TYPE_ATTACHMENT) ||
+				 Objects.equals(
+					 objectField.getDBType(),
+					 ObjectFieldConstants.DB_TYPE_CLOB) ||
+				 Objects.equals(
+					 objectField.getDBType(),
+					 ObjectFieldConstants.DB_TYPE_STRING)) {
+
+			String fieldName = "nestedFieldArray.value_text";
+
+			if (!Validator.isBlank(objectField.getIndexedLanguageId()) &&
```

Comment:
I believe this is unnecessary, since the `searchContext.getLocale()` can never be blank to match the `objectField.getIndexedLanguageId()` if it happens to be blank.

Reply:
Great, I wasn't sure about that. Thanks, I'll change this!